engine/consensus/approvals/assignment_collector.go

```diff
@@ -61,12 +65,19 @@ func NewAssignmentCollector(result *flow.ExecutionResult, state protocol.State,
 	if err != nil {
 		return nil, err
 	}
+	// pre-select all authorized verifiers at the block that is being sealed
+	authorizedApprovers, err := authorizedVerifiersAtBlock(state, result.BlockID)
+	if err != nil {
+		return nil, engine.NewInvalidInputErrorf("could not determine authorized verifiers for sealing candidate: %w", err)
+	}
```

Comment:
I just shifted this code up. Thereby, we are initializing the dependencies of `AssignmentCollector` first and subsequently the struct itself.